### PR TITLE
[Refactor](query) refactor lock in fragment mgr and change std::unorder_map to phmap

### PR DESCRIPTION
**Pull request by HappenLee**
…

### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

File: be/src/runtime/fragment_mgr.cpp [hunk 1]
```diff
@@ -602,37 +601,45 @@ Status FragmentMgr::start_query_execution(const PExecPlanFragmentStartRequest* r
...
         }
     }
     return nullptr;
 }
 
-std::shared_ptr<QueryContext> FragmentMgr::get_or_erase_query_ctx_with_lock(
-        const TUniqueId& query_id) {
-    std::unique_lock<std::mutex> lock(_lock);
-    return _get_or_erase_query_ctx(query_id);
+std::pair<Status, std::shared_ptr<QueryContext>> FragmentMgr::_get_or_insert_query_ctx(
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:626]: return  Result

**Comment by HappenLee** [comment]
run buildall

File: be/src/runtime/fragment_mgr.cpp [hunk 2]
```diff
@@ -657,8 +662,7 @@ Status FragmentMgr::_get_query_ctx(const Params& params, TUniqueId query_id, boo
     } else {
         // Find _query_ctx_map, in case some other request has already
         // create the query fragments context.
-        std::lock_guard<std::mutex> lock(_lock);
-        if (auto q_ctx = _get_or_erase_query_ctx(query_id)) {
+        if (auto q_ctx = get_query_ctx(query_id)) {
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:665]: 在并发的时候，665行，并行多个get query ctx，得到的qctx == nullptr，然后都会运行底下创建和初始化query ctx的逻辑。但是最终只有一个会加入到query ctx map 中，其他的就会自动析构。 但是我们再query ctx的析构函数里做了一些操作。 例如：    _exec_env->runtime_query_statistics_mgr()->set_query_finished(print_id(_query_id));

这些操作，都是跟query id 绑定的，此时会出现错误。

File: be/src/runtime/fragment_mgr.cpp [hunk 3]
```diff
@@ -645,8 +651,7 @@ Status FragmentMgr::_get_query_ctx(const Params& params, TUniqueId query_id, boo
     });
     if (params.is_simplified_param) {
         // Get common components from _query_ctx_map
-        std::lock_guard<std::mutex> lock(_lock);
-        if (auto q_ctx = _get_or_erase_query_ctx(query_id)) {
+        if (auto q_ctx = get_query_ctx(query_id)) {
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:654]: also rename this method _get_query_ctx  to  create_or_get_query_ctx

**Comment by HappenLee** [comment]
run buildall

**Review by zhiqiang-hhhh — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**